atomic_reactor/plugins/post_generate_maven_metadata.py

```diff
@@ -56,6 +66,101 @@ def get_nvr_components(self, nvr_requests):
 
         return components
 
+    def download_sources(self, url_requests):
+        download_queue = []
+        source_url_to_artifacts = {}
+        remote_source_files = []
+        no_source_artifacts = []
+        session = util.get_retrying_requests_session()
+
+        for url_request in url_requests:
+            artifact_checksums = {algo: url_request[algo] for algo in
+                                  hashlib.algorithms_guaranteed
+                                  if algo in url_request}
+
+            artifact = {
+                'url': url_request['url'],
+                'checksums': artifact_checksums,
+                'filename': os.path.basename(url_request['url'])
+            }
+
+            if 'source-url' not in url_request:
+                no_source_artifacts.append(artifact)
+                msg = f"No source-url found for {url_request['url']}.\n"
+                msg += 'fetch-artifacts-url without source-url is deprecated'
+                self.log.warning(msg)
+                self.log.user_warning(msg)
+                continue
+
+            source_url = url_request['source-url']
+
+            checksums = {algo: url_request[('source-' + algo)] for algo in
+                         hashlib.algorithms_guaranteed
+                         if ('source-' + algo) in url_request}
+
+            if source_url not in source_url_to_artifacts:
+                source_url_to_artifacts[source_url] = [artifact]
+                # source_url will mostly be gerrit URLs that don't have filename
+                #  in the URL itself so we'll have to get filename from URL response
+                try:
+                    target = session.get(source_url).headers.get("Content-disposition") \
```

Comment:
IS guaranteed that `Content-disposition` will be always in the response header?

Reply:
I've reversed the logic for this part. First we'll take basename from URL as default filename. If it doesn't match valid filename regex then as a last resort we'll look for a filename in headers. If it doesn't exist in headers then we'll just fail with either an AttrubuteError or IndexError since we cannot move forward without a valid filename. WDYT?